### PR TITLE
TST: change skip to xfail for test_compare_numdiff on OSX

### PR DESCRIPTION
**Pull request by jbrockmendel**
xref #5628

### CODE REVIEW

**Comment by bashtage** [comment]
This is not quite right.  

The correct solution is to run the test on all platforms in a Try, except, and then if except AssertionError is raised an and platform is OSX, then then to pytest.xfail(reason='') or .skip('').  This test passes on some OSX versions (see Travis, where is has been passing using Conda/MKL I think). 

So something like

```
# GH XXXX, YYYY explains reason why this is used
try: 
    buggy_on_osx_maybe
except AssertionError as err:
   if PLATFORM_OSX:
        pytest.xfail(reason="Known to fail on OSX")
   raise err
```

**Comment by bashtage** [comment]
https://travis-ci.org/statsmodels/statsmodels/builds/528831936 has a passing OSX 3.7

File: statsmodels/tools/_testing.py [hunk 1]
```diff
@@ -183,3 +185,24 @@ def check_predict_types(results):
         cls = pd.Series if predicted.ndim == 1 else pd.DataFrame
         predicted_expected = cls(predicted, index=exog_index)
         assert_equal(predicted_expected, predicted_pandas)
+
+
+def try_and_xfailif(condition, reason):
```
Comment (bashtage) [inline statsmodels/tools/_testing.py]: could you rename to `allowfailif` (to be testing-framework-like) of `allow_fail_if` to be Python-like.

**Comment by jbrockmendel** [comment]
@bashtage same duration permissions problem

**Comment by bashtage** [comment]
Could you rebase, squash (or skip the commit with azure changes) and remove the azure template changes from the PR.  Not sure how they ended up there, but they are no longer current and I don't want to change it.

**Pull request merged**